Multiples fixes + improvements

Geolim4 opened the pull request:
- Added CLIENT_VERSION constant
- Fixed deprecated php comments "#"
- Fixed missing/wrong PhpDoc comments
- Fixed missing function lang_get_text()
- Fixed undefined method error issueDetail by getIssue()

Geolim4 committed:
- Added CLIENT_VERSION constant
- Fixed deprecated php comments "#"
- Fixed missing/wrong PhpDoc comments
- Fixed missing function lang_get_text()
- Fixed undefined method error issueDetail by getIssue()

Geolim4 committed:
- Added $view_state parameter to addNote() method

vboctor commented on MantisPhpClient.php:
```diff
@@ -130,7 +142,7 @@ public function getMantisVersion() {
      * Gets categories list of a project.
      * @param int $p_project_id <p>Project id must be supplied</p>
-     * @return Array <p> Result array</p>
+     * @return array <p> Result array</p>
```
Please remove the <p> tags from phpdocs.

vboctor commented on MantisPhpClient.php:
```diff
@@ -295,14 +307,14 @@ public function getEnumViewState() {
 
     /**
-     * Get the list of accessible projects.
+     * @return array The list of accessible projects.
```
I generally prefer having a line the describes the function that is separate from params and return value documentation.  I think this is better than just a return tag.  I prefer the function description to focus on the logic and return tag to focus on the value type returned, array format, possible values, can be null, etc.  Please apply to all methods that don't have a function description.

vboctor commented on MantisPhpClient.php:
```diff
@@ -347,15 +363,16 @@ private function getProjectFromProjectTree( $p_project_id, $p_projects ) {
      * Checks if a user exists.
      *
-     * @param $p_user array that can contain id, name, or email.
+     * @param StdClass $p_user array that can contain id, name, or email.
```
Is it a class or an array?  The type and comment conflict.  Also when referring to StdClass/array, it would be great if we specify what properties we care about in such object/array.

vboctor commented on MantisPhpClient.php:
```diff
@@ -347,15 +363,16 @@ private function getProjectFromProjectTree( $p_project_id, $p_projects ) {
         if ( version_compare( $t_mantis_version, '1.2.16dev' ) >= 0 ) {
             try {
-                $this->soap_client->mc_project_get_issues_for_user( $this->username, $this->password, ALL_PROJECTS, 'reported', $p_user, 1, 1 );
+                $this->soap_client->mc_project_get_issues_for_user( $this->username, $this->password, 'ALL_PROJECTS', 'reported', $p_user, 1, 1 );
```
Remove the quotes around ALL_PROJECTS, and define it as a constant with value 0.

vboctor commented on MantisPhpClient.php:
```diff
@@ -347,15 +363,16 @@ private function getProjectFromProjectTree( $p_project_id, $p_projects ) {
-     * @param $p_user array that can contain id, name, or email.
+     * @param StdClass $p_user array that can contain id, name, or email.
      * @return true found, false otherwise.
+     * @throws Exception
```
Clarify when we throw an exception, e.g. when used against an old version of MantisBT.

vboctor commented on MantisPhpClient.php:
```diff
@@ -474,16 +491,29 @@ public function getCustomFieldDefinitions( $p_project_id ) {
+   * @param string $p_file_type
+   * @param string $p_file_path
+   * @return mixed
+   */
```
Indentation of phpdoc is not correct here.

vboctor commented on MantisPhpClient.php:
```diff
@@ -671,11 +712,33 @@ private function getUserPreference( $p_preference ) {
+   * @param string $str
+   * @return string
+   */
+  function lang_get_text($str){
```
Instead of defining this.  Let's just use the English values directly in getStandardFiltersList.  If the client wants localization, then can do the localization later based on filter.id or just use the English name via filter.name.

vboctor commented on MantisPhpClient.php:
```diff
@@ -10,60 +10,66 @@
+     * @param string $p_password
+     * @param string string $p_user_agent
      */
     public function __construct( $p_soap_wsdl_url, $p_username, $p_password, $p_user_agent = 'MantisPhpClient' ) {
```
Let's default $p_user_agent to null here.  Then inside the constructor if value is null, set it to MantisPhpClient/CLIENT_VERSION rather than just MantisPhpClient.

vboctor reviewed (changes requested):
Thanks @Geolim4 for your contribution.  I've reviewed the changes and provided some suggestions.

Geolim4 commented:
Ok I'll do it tonight :)

Geolim4 committed:
- Updated file header to make use of @author/@licence tags
- Added ALL_PROJECT constant (0)
- Updated all properties/method from private to protected to make extends easier
- Reviewed @returns/@params/@throws comments to be more consistent
- Modified Filter class name to MantisClientFilter due to a too generic name to prevent conflicts
- Removed redundant function lang_get_text()
- Fixed PhpDoc indentation
- Splited very long descriptions into multiple lines

Geolim4 commented:
Done :)

Geolim4 committed:
Fixed forgotten review
